sphinxcontrib/doxylink/doxylink.py

```diff
@@ -53,6 +57,24 @@ def report_warning(env, msg, docname=None, lineno=None):
     else:
         env.warn(docname, msg, lineno=lineno)
 
+def is_url(str_to_validate):
+    ''' Helper function to check if string contains URL
+
+    Args:
+        str_to_validate (str): String to validate as URL
+
+    Returns:
+        bool: True if given string is a URL, False otherwise
+    '''
+    regex = re.compile(
```

Comment:
I am against regex checking and as we can see it gets complex. I assume you looked for external options, but did not find them? I know Django is a bit overkill to drag in as dependency but as example how they have it, we could also produce something similar? https://github.com/django/django/blob/master/django/core/validators.py#L65

Reply:
I didn't find any good external options. Django uses regular expressions for validation.

Alternatively, we could use `urllib.parse.urlparse(tag_filename).scheme`, similarly as L326, but that's a bit basic.

Reply:
Since all we care about is whether the path is remote or not, it could be as simple as a `.startwith("http")`. `urlparse` would be a step up in reliability. My concern with a a "complex" regular expression is that it's easy to miss corner-cases.

However, since this is wrapped up in a single function,it's easy enough to modify the logic in the future if there's a need to a change.

Reply:
I agree that the regex seems abstract. I've added tests to show what it does in 22c226360f7485a4126c6e1fcabe39549ca03e1a.